v2/pkg/genruntime/extensions/claimer.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package extensions
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+
+	. "github.com/Azure/azure-service-operator/v2/internal/logging"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+)
+
+// Claimer can be implemented to customize how the reconciler claims a resource
```

Comment:
Do you need to doc what Claiming is about?

Reply:
Its part of the `Reconciler` interface we define and is described in that interface already:
```
	// Claim performs resource specific claim actions. This must be idempotent.
	// A standard finalizer is added to all resources, Claim
	// should deal with any resource specific claiming actions (such as setting a resource ID annotation, etc).
	// If Claim returns an error then reconciliation will be retried according to the returned Ready condition until
	// no error is returned. Once Claim succeeds CreateOrUpdate is called.
	Claim(
		ctx context.Context,
		log logr.Logger,
		eventRecorder record.EventRecorder,
		obj MetaObject) error
```

Do you think we need more than that?

Reply:
No, that'll do. I missed that.